packages/ionicons/src/index.ts

```diff
@@ -0,0 +1,20 @@
+import type { JSX as IonicJSX } from 'ionicons/dist/types/components';
```

Comment:
Are these declarations (except `export * from './icons'`) needed in the icons module? It seems to me like the module doesn't really need JSX

Reply:
It's not necessary but It allows you to use <ion-icon /> component directly like this inside JSX of Solid components.

So I can do this:
```typescript
export function Button({ children }) {
    return <button>
        <ion-icon name="square" />
        {children}
    <button />;
}
```

Sure you should use our <IonIcon>, but you can use the web component directly. It may help people to switch from something else to Solid; they can use their existing JSX/HTML.

I put the same definitions into `packages/core/src/components/index.ts`.

Reply:
Well,.. we have `<IonIcon>` in the **core** package, but it does not expose the type for `<ion-icon>`, so I added it here. But because `IonIcon` is in the **core**, the `<ion-icon>` should be also in the **core**, right?